python/cuspatial/cuspatial/geometry/geoseries.py

```diff
@@ -0,0 +1,135 @@
+    shapely-like objects. Our goal is to give feature parity with GeoPandas.
+    At this time, only from_geopandas and to_geopandas are directly supported.
+    cuspatial GIS, indexing, and trajectory functions depend on the arrays
+    stored in the `GeoArrowBuffers` object, accessible with the `points`,
+    `multipoints`, `lines`, and `polygons` accessors.
+
+    >>> cuseries.points
+        xy:
+        0   -1.0
+        1    0.0
+        dtype: float64
+    """
+
+    def __init__(
+        self,
+        data: Union[gpd.GeoSeries],
+        index: Union[cudf.Index, pd.Index] = None,
+        dtype=None,
+        name=None,
+        nan_as_null=True,
+    ):
+        # Condition index
+        if isinstance(data, (gpGeoSeries, GeoSeries)):
+            if index is None:
+                index = data.index
+        if index is None:
+            index = cudf.RangeIndex(0, len(data))
+        # Condition data
+        if isinstance(data, pd.Series):
+            data = gpGeoSeries(data)
+        # Create column
+        if isinstance(data, GeoColumn):
+            column = data
+        elif isinstance(data, GeoSeries):
+            column = data._column
+        elif isinstance(data, gpGeoSeries):
+            adapter = GeoPandasAdapter(data)
+            buffers = GeoArrowBuffers(adapter.get_geoarrow_host_buffers())
+            pandas_meta = GeoMeta(adapter.get_geopandas_meta())
+            column = GeoColumn(buffers, pandas_meta)
+        else:
+            raise TypeError(
+                f"Incompatible object passed to GeoSeries ctor {type(data)}"
+            )
+        super().__init__(column, index, dtype, name, nan_as_null)
+
+    @property
+    def _geocolumn(self):
+        """
+        The GeoColumn object keeps a reference to a `GeoArrowBuffers` object,
+        which contains all of the geometry coordinates and offsets for thie
+        `GeoSeries`.
+        """
+        return self._column
+
+    @_geocolumn.setter
+    def _geocolumn(self, value):
+        if not isinstance(value, GeoColumn):
+            raise TypeError
+        self._column = value
+
+    @property
+    def points(self):
+        """
+        Access the `PointsArray` of the underlying `GeoArrowBuffers`.
+        """
+        return self._geocolumn.points
+
+    @property
+    def multipoints(self):
+        """
+        Access the `MultiPointArray` of the underlying `GeoArrowBuffers`.
+        """
+        return self._geocolumn.multipoints
+
+    @property
+    def lines(self):
+        """
+        Access the `LineArray` of the underlying `GeoArrowBuffers`.
+        """
+        return self._geocolumn.lines
+
+    @property
+    def polygons(self):
+        """
+        Access the `PolygonArray` of the underlying `GeoArrowBuffers`.
+        """
+        return self._geocolumn.polygons
+
+    def __repr__(self):
+        # TODO: Limit the the number of rows like cudf does
+        return self.to_pandas().__repr__()
```

Comment:
Ideally we could just inherit the `__repr__` from `cudf.Series` here, no?

Reply:
That's pretty complicated right now. `cudf.Series.__repr__()` depends on `.iloc` over slices, which is not supported by `GeoSeries` yet because assembling new `GeoSeries` from multiple rows involves collecting all of the indices and slices of sub-columns that are computed in the `GpuGeometry` classes, then  copying them all into new properly sized buffers. At present only a single row item can be referenced at a time in `GeoSeries`, because our algorithms don't work on rows, but on the underlying `GeoArrow` columns.

Right now i'd rather leave this as is and can implement `GeoSeries` slicing if necessary down the road.